proxy/src/source.rs

```diff
@@ -217,7 +214,7 @@ where
         let project_name =
             ProjectName::from_string(name.to_string()).expect("project name creation failed");
         let project_domain =
-            ProjectDomain::from_string("rad".to_string()).expect("project domain creation faile");
+            ProjectDomain::from_string("rad".to_string()).expect("project domain creation failed");
```

Comment:
You could probably clean these two `let`s up using the `try_into` we just defined above, possibly?

Reply:
No, that didn't work. Throws these sorts of errors:

```
error[E0382]: use of moved value: `name`
   --> src/source.rs:234:13
    |
210 |     fn register_project(&self, name: String, description: String, img_url: String) -> Result<Project, RegistryError> {
    |                                ---- move occurs because `name` has type `std::string::String`, which does not implement the `Copy` trait
...
215 |             ProjectName::from_string(name.into()).unwrap(),
    |                                      ---- value moved here
...
234 |             name,
    |             ^^^^ value used here after move
```

and

```
error[E0277]: `?` couldn't convert the error to `substrate_subxt::error::Error`
   --> src/source.rs:215:53
    |
215 |             ProjectName::from_string(name.try_into()?).unwrap(),
    |                                                     ^ the trait `std::convert::From<std::convert::Infallible>` is not implemented for `substrate_subxt::error::Error`
```

Not knowing what to do about it, I left it as `to_string` and replaced the `.expect` calls with `.unwrap` for now, until I understand enough to apply a more robust approach.

Reply:
Ah, I might have been unclear. But I think you could roughly change this to:
```rust
let (project_name, project_domain) = ProjectId { name: name.to_string(), "rad".to_string }.try_into().expect("Failed to create registry::ProjectId");
```

That's me guessing that application's `ProjectId` constructor looks like this :sweat_smile: But basically the idea was to use the `try_into` so you're not repeating the `from_string` logic in both places :)